getnacelle/nacelle-js
# feat(starters/gatsby): add gatsby starter

### WHY are these changes introduced?

Addresses [LS-1450](https://nacelle.atlassian.net/browse/LS-1450)

> v2 Gatsby Starter: Finish up routing, all general functionality



### WHAT is this pull request doing?



#### `starters/gatsby`

This PR adds a new `starters/gatsby` project.

#### `starters/next`

- Removes the `precommit` hook, since it will not have meaning outside of the context of `nacelle-js` / in a fresh project, without additional setup.
- Adjusts the ESLint config to resolve `eslint-plugin-import` to the version used by `eslint-config-next`, rather than the top-level `nacelle-js` version.

#### `starters/nuxt-2`

Adds missing `sass` and `sass-loader` dependencies (this was causing [build failures](https://vercel.com/nacelle/nacelle-js-nuxt-starter/5MsXKpXHg11iSBzuxfax7CVR4JuX)).

#### `starters/*`

Suggests `degit` for project scaffolding. We can update this with CLI commands once the [LS-1485](https://nacelle.atlassian.net/browse/LS-1485) is complete.

#### Lerna / Monorepo-wide

Because we now have some `starters/*` projects which use Webpack 4 and others which use Webpack 5, hoisting with Lerna is breaking down. In the interest of prioritizing "things working as expected" over "low disk space usage and improved dependency install performance," this PR removes the `--hoist` flag from the `lerna boostrap` command.

### How to Test

1. Check out the `LS-1450-gatsby-starter` branch.
2. `npm run bootstrap` from the monrepo root.
3. `cd starters/gatsby`
4. Add a `.env`:
```dotenv
GATSBY_NACELLE_STOREFRONT_ENDPOINT=https://storefront.api.development.nacelle.com/graphql/v1/spaces/cky4xgh48965713nxjkv9s75uk
GATSBY_NACELLE_STOREFRONT_TOKEN=16f0d12a-7990-47dc-a5f0-4bd043e2b6e7
GATSBY_NACELLE_STOREFRONT_LOCALE=en-US

GATSBY_MYSHOPIFY_DOMAIN=pepper-wood-apparel
GATSBY_SHOPIFY_STOREFRONT_CHECKOUT_TOKEN=477e3f771eb126e58db487428cd34948
GATSBY_SHOPIFY_STOREFRONT_API_VERSION=2021-10
```
5. `npm run develop` / `npm run build`

You should be able to navigate the site all the way to checkout:

https://user-images.githubusercontent.com/5732000/151626664-de03e1b8-9ad2-45d2-87ef-2f35c728aeec.mov

### Testing checklist

* [x] Tested for accessibility (`alt` text is flowing to images)
* [x] Updated the [`README.md`](https://github.com/getnacelle/nacelle-js/tree/LS-1450-gatsby-starter/starters/gatsby#nacelle-gatsby-starter) with documentation changes
* [x] Updated public facing documentation ([link to internal doc](https://nacelle.atlassian.net/l/c/a92Q8Wxr))
